rpcserver.go

```diff
@@ -7136,8 +7152,22 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 
 	// Now that we have the accepted quote, we know the amount in Satoshi
 	// that we need to pay. We can now update the invoice with this amount.
-	mSatPerUnit := acceptedQuote.AskPrice
-	iReq.ValueMsat = int64(req.AssetAmount * mSatPerUnit)
+	//
+	// First, un-marshall the ask asset rate from the accepted quote.
+	askAssetRate, err := rfqrpc.UnmarshalFixedPoint(
+		acceptedQuote.AskAssetRate,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling ask asset rate: %w",
+			err)
+	}
+
+	// Convert the asset amount into a fixed-point.
+	assetAmount := rfqmsg.NewBigIntFixedPoint(req.AssetAmount, 0)
```

Comment:
I thought about this and what @Roasbeef said a lot, and I think I agree with him that we can't use 0 here.
Because we are going to calculate a price for the asset, it is important that we know the inherent representation of the asset. So basically if we use 0, we would imply one asset unit being 1 dollar, even though the asset might have a decimal display of 3 or 6.

So we do need to query the decimal display value here and use it. That also makes sure we have synced the issuance proof for that asset when attempting to create an invoice.

Reply:
This is how I see it:

We're calculating msat value from asset units and an asset unit to BTC rate. The decimal display is not relevant to the calculation. I've added this unit test to show that scale 0 works here: https://github.com/lightninglabs/taproot-assets/pull/1136/commits/875c0e2c1c94b8cb912a42ee91fcfdd6e1db1e43

Why is decimal display useful here for converting asset units to msats?

Reply:
Are we saying that `req.AssetAmount` is not in asset units but in real world currency units here? I think not because it's a `uint64`, right?

Reply:
Ultimately it _is_ in asset units, just unscaled. I think that's where they divergence is in our thought models here, and also where I want to make sure we're very careful (re use of constructors, etc). My fear was that if this _was_ already a scaled value (output from some other operation that didn't scale down), then we'd compound an error by not carrying along what the true scale should be. 

So far now, I think this seems to be fine, but ultimately the further unit and itests will be the true judge. 

I think my scenario above can still happen, if we ever write a scaled value to wire/rpc w/o the scale itself. Will think about how we can add some sanity checks or assertions elsewhere to make sure we catch instances like this.